Hyper/Window.cs

```diff
@@ -1,90 +1,34 @@
-
-            0.5f,  0.5f, -0.5f, 1.0f, 1.0f, // top right, back
-            -0.5f,  0.5f, -0.5f, 0.0f, 1.0f, // top left, back
-            -0.5f, -0.5f, -0.5f, 0.0f, 0.0f, // bottom left, back
-            0.5f, -0.5f, -0.5f, 1.0f, 0.0f, // bottom right, back
-
-            0.5f,  0.5f,  0.5f, 1.0f, 1.0f, // top right, top
-            -0.5f,  0.5f,  0.5f, 0.0f, 1.0f, // top left, top
-            -0.5f,  0.5f, -0.5f, 0.0f, 0.0f, // bottom left, top
-            0.5f,  0.5f, -0.5f, 1.0f, 0.0f, // bottom right, top
-
-            0.5f, -0.5f,  0.5f, 1.0f, 1.0f, // top right, bottom
-            -0.5f, -0.5f,  0.5f, 0.0f, 1.0f, // top left, bottom
-            -0.5f, -0.5f, -0.5f, 0.0f, 0.0f, // bottom left, bottom
-            0.5f, -0.5f, -0.5f, 1.0f, 0.0f, // bottom right, bottom
-
-            0.5f,  0.5f,  0.5f, 1.0f, 1.0f, // top right, right
-            0.5f, -0.5f,  0.5f, 0.0f, 1.0f, // top left, right
-            0.5f, -0.5f, -0.5f, 0.0f, 0.0f, // bottom left, right
-            0.5f,  0.5f, -0.5f, 1.0f, 0.0f, // bottom right, right
-
-            -0.5f,  0.5f,  0.5f, 1.0f, 1.0f, // top right, left
-            -0.5f, -0.5f,  0.5f, 0.0f, 1.0f, // top left, left
-            -0.5f, -0.5f, -0.5f, 0.0f, 0.0f, // bottom left, left
-            -0.5f,  0.5f, -0.5f, 1.0f, 0.0f  // bottom right, left
-        };
-
-        private readonly uint[] _indices =
-        {
-            0,  1,  3,  1,  2,  3,  // Front face
-            4,  5,  7,  5,  6,  7,  // Back face
-            8,  9, 11,  9, 10, 11,  // Top face
-            12, 13, 15, 13, 14, 15,  // Bottom face
-            16, 17, 19, 17, 18, 19,  // Right face
-            20, 21, 23, 21, 22, 23   // Left face
-        };
-
-        private int _elementBufferObject;
-
-        private int _vertexBufferObject;
+        private CancellationTokenSource _debugCancellationTokenSource = null!;
 
-        private int _vertexArrayObject;
+        private List<Object3D> _objects = null!;
 
-        private Shader _shader;
+        private Shader _shader = null!;
 
-        private Texture _texture;
+        private Texture _texture = null!;
 
-        private Texture _texture2;
+        private float _scale = 0.1f;
 
-        // The view and projection matrices have been removed as we don't need them here anymore.
-        // They can now be found in the new camera class.
-
-        // We need an instance of the new camera class so it can manage the view and projection matrix code.
-        // We also need a boolean set to true to detect whether or not the mouse has been moved for the first time.
-        // Finally, we add the last position of the mouse so we can calculate the mouse offset easily.
-        private Camera _camera;
+        private Camera _camera = null!;
 
         private bool _firstMove = true;
 
         private Vector2 _lastPos;
 
-        private double _time;
+        private Vector3 _cameraPosition = Vector3.Zero;
```

Comment:
Shouldn't this be in Camera class?

Reply:
It should. Just remember that we need to "move the world" (modify the translation matrix) instead of moving the camera